pkg/kubectl/kubectl.go

```diff
@@ -2,34 +2,14 @@ package kubectl
 
 import (
 	"fmt"
-	"os/exec"
 	"strings"
 )
 
-func executeCommand(args []string) (string, error) {
-	cmdOut, err := exec.Command("kubectl", args...).CombinedOutput()
-	if err != nil {
-		// Kubectl error messages output to stdOut
-		return "", fmt.Errorf(formatCmdOutput(cmdOut))
-	}
-	return formatCmdOutput(cmdOut), nil
-}
-
-// Execute executes kubectl <args> and returns the combined stdout/err output.
-func Execute(args ...string) (string, error) {
-	return executeCommand(args)
-}
-
-// ExecuteWithGlobalArgs is a convenience version of Execute that lets the user
-// specify global arguments as an array. Global arguments are arguments that are
-// not specific to a kubectl sub-command, eg. --kubeconfig. The list of global
-// options can be retrieved with kubectl options.
-func ExecuteWithGlobalArgs(globalArgs []string, args ...string) (string, error) {
-	return executeCommand(append(globalArgs, args...))
-}
-
-func formatCmdOutput(output []byte) string {
-	return strings.TrimPrefix(strings.TrimSuffix(strings.TrimSpace(string(output)), "'"), "'")
+// Client implements a kubectl client to execute commands
+type Client interface {
+	Execute(args ...string) (string, error)
+	ExecuteWithGlobalArgs(globalArgs []string, args ...string) (string, error)
+	IsPresent() bool
```

Comment:
As discussed yesterday, IsPresent() could probably be just a method of the local client and not part of the interface.